File: Esquenta/Entities/Produto.cs
using Esquenta.Entities.Interfaces;
using System.Collections.Generic;

namespace Esquenta.Entities
{
    public class Produto : IBaseEntity
    {
        public Produto()
        {
            Itens = new List<ProdutoItem>();
        }
        public virtual int Id { get; protected set; }
        public virtual string CodigoBarras { get; set; }
        public virtual string Nome { get; set; }
        public virtual string Descricao { get; set; }
        public virtual long Quantidade { get; set; }
        public virtual decimal Valor { get; set; }
        public virtual IList<ProdutoItem> Itens { get; set; }
    }
}

File: V2-Node/esquenta.v2/server/models/produto.js
'use strict'
module.exports = (sequelize, DataTypes) => {
  var produto = sequelize.define('produto', {
    nome: DataTypes.STRING,
    codigoBarras: DataTypes.STRING,
    descricao: DataTypes.STRING,
    quantidade: DataTypes.INTEGER,
    quantidadeMinima: DataTypes.INTEGER,
    valor: DataTypes.DECIMAL
  }, {})
  produto.associate = function (models) {
    produto.hasMany(models.produtoItem, {
      foreignKey: 'parentId',
      as: 'Items'
    })
  }
  return produto
}
